addons/common/functions/fnc_exportMissionSQF.sqf

```diff
@@ -0,0 +1,381 @@
+#include "script_component.hpp"
+/*
+ * Author: mharis001
+ * Returns SQF code that can be executed to restore the current mission.
+ *
+ * Arguments:
+ * 0: Position <ARRAY> (default: [0, 0, 0])
+ * 1: Radius <NUMBER> (default: -1)
+ *   - in meters, -1 for entire mission
+ * 2: Include Waypoints <BOOL> (default: true)
+ * 3: Include Markers <BOOL> (default: false)
+ * 4: Add To Curators <BOOL> (default: false)
+ *
+ * Return Value:
+ * Mission SQF <STRING>
+ *
+ * Example:
+ * [] call zen_common_fnc_exportMissionSQF
+ *
+ * Public: No
+ */
+
+#define NEWLINE toString [ASCII_NEWLINE]
+
+#define VAR_OBJECT(index) format ["_object%1", index]
+#define VAR_GROUP(index) format ["_group%1", index]
+
+params [
+    ["_position", [0, 0, 0], [[]], [2, 3]],
+    ["_radius", -1, [0]],
+    ["_includeWaypoints", true, [true]],
+    ["_includeMarkers", false, [true]],
+    ["_addToCurators", false, [true]]
+];
+
+private _processedObjects = [];
+private _processedGroups  = [];
+
+private _outputGroups1   = [];
```

Comment:
Ew, bad names.

Reply:
Not sure what would be better, open to suggestions.